dist/performance.js

```diff
@@ -84,6 +84,66 @@ function getWebVitalsJS() {
     }).map(har => har.url);
 }
 
+function getGamingMetrics(rawDoc) {
+    let returnObj = {};
+    const regexForCheckChromeLH = new RegExp(/.{1}userAgent.{1,100}(?:Chrome-Lighthouse|Google Lighthouse).{2}/)
+    const regexForCheckGTmetrix = new RegExp(/.{1}userAgent.{1,100}(?:GTmetrix|gtmetrix.com).{2}/)
+    const regexForCheckPageSpeed = new RegExp(/.{1}userAgent.{1,100}(?:PageSpeed).{2}/)
+    let scripts = rawDoc.getElementsByTagName('script');
+    for (let i = 0; i < scripts.length; i++) {
+        if (scripts[i].src) {
+            continue;
+        }
+        let scriptTagCode = scripts[i].innerHTML;
+        if (regexForCheckChromeLH.test(scriptTagCode)){
+            returnObj['detectUA-ChromeLH'] = true;
+        }
+        if(regexForCheckGTmetrix.test(scriptTagCode)){
+            returnObj['detectUA-GTmetrix'] = true;
+        }
+        if(regexForCheckPageSpeed.test(scriptTagCode)){
+            returnObj['detectUA-PageSpeed'] = true;
+        }
+    }
+
+    //https://www.debugbear.com/blog/optimizing-web-vitals-without-improving-performance
+    //catch lcp animation / cls animation & overlay hack
+    const regexForCheckfadeInAnimation = new RegExp(/this.style.animation.{1,10}.fadein.{1,20}.forwards/)
```

Comment:
@mel-ada this regex is pretty specific to the sample code in the post. Do you want to only measure sites that adopt that exact code, or loosen the detection up to try to capture instances where we think it's likely they're using these hacks?

Like, if the image's opacity is 0 and there's an onload attribute, they're already doing something weird.

Reply:
Good point - since we already have this metric let's keep it. Let's also add a check for the more generic:
`lcpImgAnimationSoft = opacity: 0 && has onload attribute`